src/utils/helpers/timezone_magic/time-based-on-timezone.js

```diff
@@ -1,4 +1,4 @@
-import timeZoneFromBrowser from 'utils/helpers/time-zone-from-browser'
+import timeZoneFromBrowser from 'utils/helpers/timezone_magic/time-zone-from-browser'
 
 const timeBasedOnTimeZone = (date, locale = 'en-US') =>
```

Comment:
I think we should initialise the locale from the locale we get from the browser, the same way we have initialised timezone.